bundles/org.openhab.ui/web/src/components/widgets/layout/oh-canvas-layout.vue

```diff
@@ -0,0 +1,260 @@
+      screenWidth: Number,
+      screenHeight: Number,
+      fullscreen: this.$fullscreen.getState(),
+      navbarHidden: false,
+      style: {
+        width: Number,
+        height: Number,
+        scale: 1.0
+      },
+      grid: {
+        pitch: Number,
+        enable: false
+      }
+    }
+  },
+  created () {
+    if (this.config.layoutType === 'fixed' && this.config.fixedType === 'canvas') {
+      this.style.width = this.screenWidth = this.config.screenWidth || 1280
+      this.style.height = this.screenHeight = this.config.screenHeight || 720
+      this.grid.pitch = this.config.grid || 5
+      this.grid.enable = this.config.gridEnable || false
+
+      if (!this.context.editmode) {
+        window.addEventListener('resize', this.setDimensions)
+      } else {
+        window.addEventListener('keydown', this.doCommand)
+        console.log('Adding keydown window listener')
+      }
+    }
+
+    this.canvasLayoutStyle()
+    this.computeLayout()
+  },
+  mounted () {
+    // Chrome reports a wrong size in fullscreen, store initial resolution and use non-dynamically.
+    this.windowWidth = window.screen.width
+    this.windowHeight = window.screen.height
+  },
+  destroyed () {
+    window.removeEventListener('keydown', this.doCommand)
+  },
+  methods: {
+    isRetina () {
+      return window.devicePixelRatio > 1
+    },
+    getCurrentScreenResolution () {
+      return (
+        'Layout Size: ' +
+        this.screenWidth +
+        ' x ' +
+        this.screenHeight +
+        ' (Current Screen: ' +
+        this.windowWidth +
+        ' x ' +
+        this.windowHeight +
+        ')'
+      )
+    },
+    addItem () {
+      this.context.component.slots['canvas'].push({
+        component: 'oh-canvas-item',
+        config: { x: 20, y: 20, h: 150, w: 200 },
+        slots: { default: [] }
+      })
+      this.computeLayout()
+    },
+    toggleFullscreen () {
+      this.$fullscreen.toggle(this.$refs.ohCanvasLayout, {
+        wrap: false,
+        callback: (fullscreen) => {
+          this.canvasLayoutStyle()
+          this.fullscreen = fullscreen
+        }
+      })
+    },
+    toggleGrid () {
+      this.context.component.config.gridEnable = this.grid.enable = !this.grid.enable
+    },
+    canvasLayoutStyle: function () {
+      if (this.config.scale && !this.context.editmode) {
+        this.style.scale = parent.innerWidth / this.screenWidth
+      } else {
+        this.style.scale = 1.0
+      }
+    },
+    computeLayout () {
+      let layout = []
+      if (this.context.component.slots?.canvas) {
+        this.context.component.slots.canvas.forEach((item) => {
+          layout.push({
+            item: item,
+            id: Math.random().toString(36).substring(2)
+          })
+        })
+      }
+      this.layout = layout
+    },
+    doCommand (e) {
+      let cmd = String.fromCharCode(e.keyCode).toLowerCase()
+      console.log(`Received key press code:${e.keyCode} string:${cmd}`)
+    }
```

Comment:
I suppose this is not necessary anymore?

Reply:
Keyboard shortcuts in edition will be in a later version...